test/cljs/chronograph/pages/organization/events_test.cljs

```diff
@@ -12,29 +12,68 @@
-               :slug       "a-test-org"
-               :created-at "2020-09-14T14:16:06.402873Z"
-               :updated-at "2020-09-14T14:16:06.402873Z"}
-              @(rf/subscribe [::subs/organization slug]))
-           "The fetched organization should be in the DB"))))
+     (let [fetch-org-event   (tu/stub-event ::org-events/fetch-organization)
+           fetch-tasks-event (tu/stub-event ::org-events/fetch-tasks)]
+       (tu/set-token (routes/path-for :organization-show :slug "test-slug"))
+       (rf/dispatch [::org-events/organization-page-mounted])
+       (is (= [::org-events/fetch-organization "test-slug"]
+              @fetch-org-event))
+       (is (= [::org-events/fetch-tasks "test-slug"]
+              @fetch-tasks-event))))))
+
+(deftest fetch-organization-test
+  (testing "When the API call for fetch organization succeeds"
+    (testing "When the user is a member of the organization"
+      (rf-test/run-test-sync
+       (tu/initialize-db!)
+       (let [slug             "a-test-org"
+             organization     {:id         42
+                               :name       "A Test Org"
+                               :slug       slug
+                               :role       "member"
+                               :created-at "2020-09-14T14:16:06.402873Z"
+                               :updated-at "2020-09-14T14:16:06.402873Z"}
+             dispatched-event (tu/stub-event ::org-events/fetch-members)]
+         (rf/reg-fx :http-xhrio
+           (fn [_]
+             (rf/dispatch [::org-events/fetch-organization-success
+                           organization])))
+         (tu/set-token (routes/path-for :organization-show :slug slug))
+         (rf/dispatch [::org-events/fetch-organization slug])
+         (is (= {:id         42
+                 :name       "A Test Org"
+                 :slug       "a-test-org"
+                 :role       "member"
+                 :created-at "2020-09-14T14:16:06.402873Z"
+                 :updated-at "2020-09-14T14:16:06.402873Z"}
+                @(rf/subscribe [::subs/organization slug]))
+             "The fetched organization should be in the DB")
+         (is (= nil
+                @dispatched-event)
+             "The members should not be fetched"))))
+
+    (testing "When the user is an admin of the organization"
+      (rf-test/run-test-sync
+       (tu/initialize-db!)
+       (let [slug             "a-test-org"
+             organization     {:id   42
+                               :name "A Test Org"
+                               :slug slug
+                               :role "admin"}
+             dispatched-event (tu/stub-event ::org-events/fetch-members)]
+         (rf/reg-fx :http-xhrio
+           (fn [_]
+             (rf/dispatch [::org-events/fetch-organization-success
+                           organization])))
+         (tu/set-token (routes/path-for :organization-show :slug slug))
+         (rf/dispatch [::org-events/fetch-organization slug])
```

Comment:
Worth checking ::subs/organization subscription to see that it is what it should be here.

Reply:
Addressed in the latest patchset